engine/consensus/dkg/reactor_engine.go

```diff
@@ -87,49 +119,85 @@ func (e *ReactorEngine) Done() <-chan struct{} {
 	return e.unit.Done()
 }
 
-// EpochSetupPhaseStarted handles the EpochSetupPhaseStared protocol event. It
-// starts a new controller for the epoch and registers the triggers to regularly
-// query the DKG smart-contract and transition between phases at the specified
-// views.
+// EpochSetupPhaseStarted handles the EpochSetupPhaseStarted protocol event by
+// starting the DKG process.
 func (e *ReactorEngine) EpochSetupPhaseStarted(currentEpochCounter uint64, first *flow.Header) {
-	firstID := first.ID()
+	e.startDKGForEpoch(currentEpochCounter, first)
+}
+
+// EpochCommittedPhaseStarted handles the EpochCommittedPhaseStarted protocol
+// event by checking the consistency of our locally computed key share.
+func (e *ReactorEngine) EpochCommittedPhaseStarted(currentEpochCounter uint64, _ *flow.Header) {
+	e.handleEpochCommittedPhaseStarted(currentEpochCounter)
+}
 
-	lg := e.log.With().
+// startDKGForEpoch starts the DKG instance for the given epoch, only if we have
+// never started the DKG during setup phase for the given epoch. This allows consensus nodes which
+// boot from a state snapshot within the EpochSetup phase to run the DKG.
+//
+// It starts a new controller for the epoch and registers the triggers to regularly
+// query the DKG smart-contract and transition between phases at the specified views.
+//
+func (e *ReactorEngine) startDKGForEpoch(currentEpochCounter uint64, first *flow.Header) {
+
+	firstID := first.ID()
+	nextEpochCounter := currentEpochCounter + 1
+	log := e.log.With().
 		Uint64("current_epoch", currentEpochCounter).
```

Comment:
better to include the `nextEpochCounter` in the log

Reply:
https://github.com/onflow/flow-go/pull/1721/commits/7c52c9c3618d2f890edae4cb9125dcb203201150